R/zchunk_L2322.Fert.R

```diff
@@ -76,272 +76,447 @@ module_energy_L2322.Fert_DISABLED <- function(command, ...) {
+      mutate(year = as.integer(year)) ->
+      A322.globaltech_coef_long
+
+    df_years <- unique(A322.globaltech_coef_long$year)
+    int_years <- setdiff(c(BASE_YEARS, FUTURE_YEARS), df_years)
+
+    A322.globaltech_coef_long %>%
+      select(-coefficient, -year) %>%
+      unique() %>%
+      repeat_add_columns(tibble(year = int_years)) %>%
+      bind_rows(A322.globaltech_coef_long) %>%
+      group_by(supplysector, subsector, technology, minicam.energy.input) %>%
+      mutate(coefficient = approx_fun(year, coefficient, rule = 1)) %>%
+      mutate(coefficient = round(coefficient, energy.DIGITS_COEFFICIENT)) %>%
+      ungroup %>%
+      filter(year %in% c(BASE_YEARS, FUTURE_YEARS)) %>%
+      rename(sector.name = supplysector, subsector.name = subsector) ->
+      L2322.GlobalTechCoef_Fert
+
+    # Costs of global technologies
+    # L2322.GlobalTechCost_Fert: Non-energy costs of global fertilizer manufacturing technologies
+    L2322.GlobalTechCoef_Fert %>%
+      select(LEVEL2_DATA_NAMES[["GlobalTechYr"]]) %>%
+      mutate(minicam.non.energy.input = "non-energy") %>%
+      left_join(L1322.Fert_NEcost_75USDkgN_F, by = c('technology' = 'fuel') ) %>% # expecting NAs in the joined tibble
+      rename(input.cost = NEcost_75USDkgN) %>%
+      mutate(input.cost = round(input.cost, energy.DIGITS_COST)) %>%
+      na.omit -> #Export technologies have no cost assigned. Just drop the object
+      L2322.GlobalTechCost_Fert
+
+    # Carbon capture rates from technologies with CCS
+    # L2322.GlobalTechCapture_Fert: CO2 capture fractions from global fertilizer production technologies with CCS
+    ## No need to consider historical periods or intermittent technologies here
+    A322.globaltech_co2capture %>%
+      gather(year, remove.fraction, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.numeric(year)) ->
+      A322.globaltech_co2capture_long
+
+    df_years <- unique(A322.globaltech_co2capture_long$year)
+    int_years <- setdiff(FUTURE_YEARS, df_years)
 
+    A322.globaltech_co2capture_long %>%
+      select(-remove.fraction, -year) %>%
+      unique() %>%
+      repeat_add_columns(tibble(year = int_years)) %>%
+      bind_rows(A322.globaltech_co2capture_long) %>%
+      group_by(supplysector, subsector, technology) %>%
+      mutate(remove.fraction = approx_fun(year, remove.fraction, rule = 1)) %>%
+      mutate(remove.fraction = round(remove.fraction, energy.DIGITS_REMOVE.FRACTION)) %>%
+      ungroup %>%
+      filter(year %in% FUTURE_YEARS) %>%
+      rename(sector.name = supplysector, subsector.name = subsector) %>%
+      mutate(storage.market = "carbon-storage") ->
+      L2322.GlobalTechCapture_Fert
+
+    # Retirement information
+    A322.globaltech_retirement %>%
+      set_years %>%
+      mutate(year = as.numeric(year)) %>%
+      rename(sector.name = supplysector, subsector.name = subsector) ->
+      A322.globaltech_retirement_with_years
+
+    # Copy the data in the first future period through to the end year
```

Comment:
The pipeline below is getting the max of the base years, not the first future period. Do you man copy the last historical year to the future years?

Reply:
Comment corrected.